File: app/src/main/java/com/yijian/staff/mvp/course/timetable/schedule/week/addstudent/step2/CourseTimeBean.java
package com.yijian.staff.mvp.course.timetable.schedule.week.addstudent.step2;

import android.support.annotation.NonNull;

/**
 * author：李帅华
 * email：850716183@qq.com
 * time: 2018/8/17 19:21:46
 */
public class CourseTimeBean implements Comparable<CourseTimeBean> {
    private String startTime;
    private String endTime;

    public String getStartTime() {
        return startTime;
    }

    public void setStartTime(String startTime) {
        this.startTime = startTime;
    }

    public String getEndTime() {
        return endTime;
    }

    public void setEndTime(String endTime) {
        this.endTime = endTime;
    }

    @Override
    public int compareTo(@NonNull CourseTimeBean o) {
        return
                Integer.parseInt(this.getStartTime().replaceAll(":", "")) - Integer.parseInt(o.getStartTime().replaceAll(":", ""));


    }
}

File: app/src/main/java/com/yijian/staff/bean/CourseTimeBean.kt
package com.yijian.staff.bean

/**
 * author：李帅华
 * email：850716183@qq.com
 * time: 2018/8/17 19:21:46
 */
class CourseTimeBean(val startTime: String,val endTime: String) : Comparable<CourseTimeBean>{


    override fun compareTo(other: CourseTimeBean): Int {
        return Integer.parseInt(this.startTime.replace(":".toRegex(), "")) - Integer.parseInt(other.startTime.replace(":".toRegex(), ""))
    }
}